Purchases/Identity/CustomerInfoManager.swift

```diff
@@ -126,8 +126,19 @@ class CustomerInfoManager {
 
         do {
             let maybeInfoDict = try JSONSerialization.jsonObject(with: customerInfoData) as? [String: Any]
-            guard let customerInfoDict = maybeInfoDict,
-                  let info = CustomerInfo(data: customerInfoDict) else {
+            guard let customerInfoDict = maybeInfoDict else {
+                return nil
+            }
+
+            let info: CustomerInfo
+            do {
+                info = try CustomerInfo(data: customerInfoDict)
+            } catch {
+                if let customerInfoError = error as? CustomerInfoError {
+                    Logger.error(customerInfoError.description)
+                } else {
+                    Logger.error("Error loading customer info from cache: \(error)")
```

Comment:
This would happen if the cached data is corrupted right? I doubt it happens often but who knows... Maybe if someone modifies the UserDefaults and breaks the JSON?

Reply:
Yeah, same feeling here. I think you're right.